### PR TITLE
Added Treetap

### PR DESCRIPTION
**Pull request by DaFuqs**
This PR adds a fully functional Treetap block, with block, item, recipe, block entity, renderer & recipe type.

Further Ideas:
- The visuals are a tad barebones and IMO the tap would benefit from a 2D inventory sprite instead of the 3D-model
- Add more recipes
- The recipe type currently only supports only single blocks as criterion, block tags, or a location predicate would be cooler.
- The recipe type does not make use of the Fabric Fluid API yet

## Summary by CodeRabbit

- **New Features**
    - Introduced new block types and a block entity for enhanced gameplay.
    - Added a new crafting recipe and its serializer for game interaction.
    - Implemented rendering capabilities for a specific block entity.

- **Refactor**
    - Updated method access levels to improve code usability.

### CODE REVIEW

**Review by MBatt1 — CHANGES_REQUESTED** [review]
The other suggestions are good and I think especially block tags and predicates (like fishing recipes do) are things we should add. 

I definitely think there should be a time gap between placing and filling, so that you can't just get infinite fluid super quickly, just a specifier in the recipe for the delay for how long it takes to 'recharge' the block, or how long it takes to fill an item

File: src/main/java/net/id/paradiselost/recipe/ParadiseLostRecipeTypes.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	static <S extends RecipeSerializer<T>, T extends Recipe<?>> S registerSerializer(String id, S serializer) {
+		return Registry.register(Registry.RECIPE_SERIALIZER, ParadiseLost.locate(id), serializer);
+	}
+
+	static <T extends Recipe<?>> RecipeType<T> registerRecipeType(String id) {
+		return Registry.register(Registry.RECIPE_TYPE, ParadiseLost.locate(id), new RecipeType<T>() {
+			@Override
+			public String toString() {
+				return "spectrum:" + id;
```
Comment (MBatt1) [inline src/main/java/net/id/paradiselost/recipe/ParadiseLostRecipeTypes.java]: 🤨

**Comment by DaFuqs** [comment]
> I definitely think there should be a time gap between placing and filling, so that you can't just get infinite fluid super quickly, just a specifier in the recipe for the delay for how long it takes to 'recharge' the block, or how long it takes to fill an item

Currently it's based on random tick to add some natural randomness to the tap. But I could also move it so scheduled ticks, always taking a fixed, predetermined amount of time.

**Comment by MBatt1** [comment]
> > I definitely think there should be a time gap between placing and filling, so that you can't just get infinite fluid super quickly, just a specifier in the recipe for the delay for how long it takes to 'recharge' the block, or how long it takes to fill an item
> 
> Currently it's based on random tick to add some natural randomness to the tap. But I could also move it so scheduled ticks, always taking a fixed, predetermined amount of time.

I just realized that I loaded a world that I think had random tick speed set to 3000, so nevermind. blocktags and predicates still sound like they'd be cool tho

**Review by MBatt1 — DISMISSED** [review]

**Comment by DaFuqs** [comment]
Dafuq

coderabbitai[bot] ***@***.***> schrieb am Mo., 11. März 2024,
18:55:

**Pull request merged**